dateformatterutil/src/main/java/com/example/dateformatter/DateFormatterUtil.kt

```diff
@@ -0,0 +1,182 @@
+        return try {
+            SimpleDateFormat(format).format(from)
+        } catch (e: ParseException) {
+            e.localizedMessage
+        }
+    }
+
+    /**
+     * @usage this function is used to convert date of type Date to hijri date String
+     * @param from A Date containing the date object
+     * @return A LocalDate object that represents the hijri date
+     */
+    fun convertToHijriDate(from: Date): LocalDate? {
+        try {
+            val calendar = Calendar.getInstance()
+            calendar.timeInMillis = from.time
+            val iso: Chronology = GregorianChronology.getInstance()
+            val hijri: Chronology = IslamicChronology.getInstance()
+
+            val todayIso = LocalDate(
+                calendar[Calendar.YEAR],
+                calendar[Calendar.MONTH] + 1,
+                calendar[Calendar.DAY_OF_MONTH], iso
+            )
+            val todayHijri = LocalDate(
+                todayIso.toDateTimeAtStartOfDay(),
+                hijri
+            )
+            return todayHijri
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+        return null
+    }
+
+    /**
+     * @usage this function is used to convert LocalDateTime object to hijri date String
+     * @param from A LocalDateTime containing the datetime object
+     * @return A String that represents the date
+     */
+    fun convert(date: LocalDateTime, format: String): String {
+        return date.toString(format)
+    }
+
+    /**
+     * @usage this function is used to convert LocalDateTime object to hijri date String
+     * @param localDate A LocalDate containing the date object
+     * @param isIslamic A Boolean confirm that the date object
+     * @return A Date that represents the localDate is islamic or normal date
+     */
+    fun convertFromIslamic(localDate: LocalDate, isIslamic: Boolean): Date? {
+        return if (isIslamic) {
+            val iso: Chronology = ISOChronology.getInstanceUTC()
+            val dtIso = localDate.toDateTimeAtStartOfDay(DateTimeZone.UTC)
+                .withChronology(iso).toLocalDate()
+            println("hijri-to-gregorian: $localDate=>$dtIso")
+            dtIso.toDate()
+        } else {
+            localDate.toDate()
+        }
+    }
+    /**
+     * @usage this function is used to convert date to hours and minutes based on UTC time zone
+     * @param date A String containing the date
+     * @return A String that represents the hours and minutes
+     */
+    fun convertDateToHoursMinUTC(dateString: String): String {
+        val format = SimpleDateFormat("HH:mm")
+        format.timeZone = TimeZone.getTimeZone("UTC")
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        var d: Date? = null
+        try {
+            d = sdf.parse(dateString)
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+        return format.format(d)
+    }
```

Comment:
Refactor this function to be more usable as from the function body it is seen that user can't figure out he must pass a dateString with specific format that the function accepts "yyyy-MM-dd'T'HH:mm:ss"